code-challenges.test.js

```diff
@@ -10,52 +10,69 @@
 // Reminder: The test will call your function
 // Run the file with the following command: $ yarn jest
 
-
 // --------------------1) Create a function that takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.
 
 // a) Create a test with expect statements using the variables provided.
 
-const secretCodeWord1 = "Lackadaisical"
+const secretCodeWord1 = "Lackadaisical";
 // Expected output: "L4ck4d41s1c4l"
-const secretCodeWord2 = "Gobbledygook"
+const secretCodeWord2 = "Gobbledygook";
 // Expected output: "G0bbl3dyg00k"
-const secretCodeWord3 = "Eccentric"
+const secretCodeWord3 = "Eccentric";
 // Expected output: "3cc3ntr1c"
 
+// a describe method that lists the name of the function OR naming of the particular test.
+describe("strReplace", () => {
+  // a test/it method, nested within the describe block, that in plain words, describes that the function does.
+  it("Will replace the letter A with the number four using RegExp", () => {
+    expect(strReplace(secretCodeWord1)).toEqual("L4ck4d41s1c4l");
+    expect(strReplace(secretCodeWord2)).toEqual("G0bbl3dyg00k");
+    expect(strReplace(secretCodeWord3)).toEqual("3cc3ntr1c");
+  });
+});
 
-write a code that took a number an dreplace it with a number
-// b) Create the function that makes the test pass.
+//write a code that took a number and replace it with a number
 
+// b) Create the function that makes the test pass.
 
+const codeWord = (string) => {
+  let strReplace = string
+    .replace(/a/gi, 4)
+    .replace(/e/gi, 3)
+    .replace(/i/gi, 1)
+    .replace(/o/gi, 0);
 
+  return string;
+};
 // --------------------2) Create a function that takes in an array of words and a single letter and returns all the words that contain that particular letter.
 
 // a) Create a test with expects statement using the variable provided.
 
-const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
-const letterA = "a"
+const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"];
+const letterA = "a";
 // Expected output: ["Apple", "Banana", "Orange"]
-const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
-const letterE = "e"
+const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"];
+const letterE = "e";
 // Expected output: ["Cherry", "Blueberry", "Peach"]
 
-
-
 // b) Create the function that makes the test pass.
 
-
+const filterWord = (array) => {
+  let newArray=array
+const result = words.filter(word => word.include(a))
+  return array;
```

Comment:
Your filter logic looks good. But then it looks like you are just returning the original array. I think calling your filter logic on your `array` parameter will get you what you need. `return array.filter(word => word.include(a))`